exec/java-exec/src/main/java/org/apache/drill/exec/cache/VectorSerializer.java

```diff
@@ -40,20 +52,18 @@
    */
 
   public static class Writer {
+    static final MetricRegistry metrics = DrillMetrics.getRegistry();
+    static final String WRITER_TIMER = MetricRegistry.name(VectorAccessibleSerializable.class, "writerTime");
 
-    private final OutputStream stream;
-    private final BufferAllocator allocator;
-    private boolean retain;
+    private final SpillSet spillSet;
+    private final WritableByteChannel channel;
+    private final OutputStream output;
     private long timeNs;
 
-    public Writer(BufferAllocator allocator, OutputStream stream) {
-      this.allocator = allocator;
-      this.stream = stream;
-    }
-
-    public Writer retain() {
-      retain = true;
-      return this;
+    private Writer(SpillSet spillSet, String path) throws IOException {
```

Comment:
Before this class was independent of the `SpillSet`. Now, it can be used only with that class. Not sure if this is an improvement.

The only thing it seems we need from `SpillSet` is to update the write byte count. Perhaps define an interface here with just the needed methods. Then, let `SpillSet` implement that interface. That gets what you need without introducing extra dependencies.

Reply:
The `Writer` in `VectorSerializer` is used only for spilling.  I'd suggest moving `VectorSerializer` to the `spill`package to make it more explicit and keep the dependency on the `SpillSet`. I am not sure if there is a need for generic `Writer` that can handle multiple use cases. I think that `Writer` needs to be optimized to handle spill use case.

`SpillSet` is also needed to get `WritableByteChannel` and encapsulates what `Writer` uses from operators.

Reply:
The `Writer` is a wrapper to make `VectorSerializer` easier to use. It is *presently* only used in spilling (that's the project that created it.) But, the intention is for it to be general purpose. By binding int to `SpillSet`, it then becomes the easy-to-use interface for spilling, but other code must use the clunkier prior interface.

Not a big issue; just trying to pass along the design intent when the code was created...